src/containerapp/azext_containerapp/custom.py

```diff
@@ -963,6 +979,41 @@ def update_containerapp_logic(cmd,
         handle_raw_exception(e)
 
 
+def update_container_app_source(cmd, containerapp_def, name, target_port, image, workload_profile_name, ingress, source, registry_server, registry_user, registry_pass):
+        from ._up_utils import (ContainerApp, ResourceGroup, ContainerAppEnvironment, _reformat_image, get_token, _has_dockerfile, _get_dockerfile_content, _get_ingress_and_target_port, _get_registry_details)
+
+        dockerfile = "Dockerfile"
+
+        # Parse resource group name and managed env name
+        env_id = containerapp_def["properties"]['environmentId']
+        parsed_managed_env = parse_resource_id(env_id)
+        env_name = parsed_managed_env['name']
+        env_rg = parsed_managed_env['resource_group']
+
+        # Set image to None if it was previously set to the default image (case where image was not provided by the user) else reformat it
+        image = None if image is None else _reformat_image(source=source, image=image, repo=None)
+        location = containerapp_def["location"]
+
+        if _has_dockerfile(source, dockerfile):
+            dockerfile_content = _get_dockerfile_content(repo=None, branch=None, token=None, source=source, context_path=None, dockerfile=dockerfile)
+            ingress, target_port = _get_ingress_and_target_port(ingress, target_port, dockerfile_content)
+
+        # Construct ContainerApp
+        resource_group = ResourceGroup(cmd, env_rg, location=location)
+        env = ContainerAppEnvironment(cmd, env_name, resource_group, location=location)
+        app = ContainerApp(cmd=cmd, name=name, resource_group=resource_group, image=image, env=env, target_port=target_port, workload_profile_name=workload_profile_name, ingress=ingress, registry_server=registry_server, registry_user=registry_user, registry_pass=registry_pass)
```

Comment:
nit: out of curiosity, do we need to reconstruct this entire Container App object or could we do something like `self.client.show` to fetch the existing Container App object that we're trying to update? Is this to handle the case where the user would want to update specific properties of the existing Container App along with the image that's produced from the source?

Reply:
Unfortunately, we cannot use `self.client.show` to fetch the existing Container App Object in this case because we are re-using methods like [_create_github_action](https://github.com/snehapar9/azure-cli-extensions/blob/snehapar/Add-source-to-containerapp-update/src/containerapp/azext_containerapp/containerapp_decorator.py#L668) (`--repo`) and [run_acr_build](https://github.com/snehapar9/azure-cli-extensions/blob/snehapar/Add-source-to-containerapp-update/src/containerapp/azext_containerapp/_up_utils.py#L585) (`--source`) that requires a [ContainerApp](https://github.com/snehapar9/azure-cli-extensions/blob/snehapar/Add-source-to-containerapp-update/src/containerapp/azext_containerapp/_up_utils.py#L346) object